math/src/elliptic_curve/short_weierstrass/curves/bls12_377/field_extension.rs

```diff
@@ -292,6 +298,51 @@ impl FieldElement<Degree12ExtensionField> {
         ])
     }
 }
+impl HasQuadraticNonResidue<Degree2ExtensionField> for LevelTwoResidue {
+    fn residue() -> FieldElement<Degree2ExtensionField> {
```

Comment:
Couldn't this be a `const fn`?

Reply:
is it possible to declare a const fn inside a trait imp?

Reply:
You're right, didn't notice that this is a trait's function. Nvm then.